### PR TITLE
Update docker tutorial

### LINKED ISSUE #830: `Microservices hosted in Docker` steps seem out of date
The [Load into Docker](https://github.com/dotnet/core-docs/blob/master/docs/csharp/tutorials/microservices.md#load-into-docker) of the [Microservices hosted in Docker](https://github.com/dotnet/core-docs/blob/master/docs/csharp/tutorials/microservices.md) seem out of date and a little heavy handed.

1. The section starts by going over the contents of the Dockerfile created by the asp.net template.  A lot of this content appears to have changed.  The current Dockerfile is not based on microsoft/dotnet:onbuild.  It doesn't load SQLite.

 Here is what the Dockerfile looks like that yo generated for me.
 '''
 FROM microsoft/dotnet:latest

 COPY . /app

 WORKDIR /app

 RUN ["dotnet", "restore"]
 
 RUN ["dotnet", "build"]
 
 EXPOSE 5000/tcp

 ENTRYPOINT ["dotnet", "run", "--server.urls", "http://0.0.0.0:5000"]
 '''

2. I don't see the reason to create a new docker machine called 'weather-service'.  This seems heavy handed and isn't necessary.

Aside from that it may be useful to include a step to attach to the running container so that the user can inspect the log output.  This can be useful when diagnosing problems on why the site is not responding.

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #830

The updated version of the tutorial uses the latest yeoman generated templates. Its also updated to use the Docker for Windows and/or Docker for Mac installations.

/cc @michaelsimons @spboyer

### CODE REVIEW

File: docs/csharp/tutorials/microservices.md [hunk 1]
```diff
@@ -354,125 +358,130 @@ the machine parameters when you start, you only need to
...
 ```
-EXPOSE 80
-ENTRYPOINT ["dotnet", "run"]
+EXPOSE 5000/tcp
+ENTRYPOINT ["dotnet", "run", "--server.urls", "http://0.0.0.0:5000"]
 ```
 
-Notice that this Dockerfile uses the dotnet cli to build and run your docker image.
-That's why the larger image is needed.
+The Dockerfile uses declares port 5000 here because it's the default port that an
```
Comment (svick) [inline docs/csharp/tutorials/microservices.md]: Uses or declares? I think you should choose only one of the two verbs.

Comment (spboyer) [inline docs/csharp/tutorials/microservices.md]: Maybe something like: Using the EXPOSE command informs Docker to listen on port 5000. This is the default port for ASP.NET Core applications and must be defined in the ENTRYPOINT command using the `--server.urls` command line option.  Alternatively you may also define this in the `WebHostBuilder.UseUrls('http://0.0.0.0:5000')` method.

File: docs/csharp/tutorials/microservices.md [hunk 2]
```diff
@@ -354,125 +358,130 @@ the machine parameters when you start, you only need to
...
+match for your container to listen on the proper port.
 
-Here are the steps to build the image and deploy it. The information below is 
-for the PowerShell CLI. Different shells will have slightly different syntax
-that will be highlighted below.
+## Building and running the image
 
-First, you have to create a new docker machine called 'weather-service':
+Let's build an image and run the service inside a Docker container. You build a container
+using the docker build command. Run the following command from the directory containing your code.
```
Comment (MichaelSimons) [inline docs/csharp/tutorials/microservices.md]: <nit> Here it says "you build a container using the docker build command".  Later on the article says "When this command completes, you have a container ready to run...".  This wording is incorrect IMO and may cause some confusion.  When you build with Docker you technically produce an image.  When you run an image, you create a container.  If you want to think of it in terms of programming, an image is a class whereas a container is an instance of a class.  Here is an older article that explains the differences (first part of the article) - [http://paislee.io/how-to-automate-docker-deployments/](http://paislee.io/how-to-automate-docker-deployments/)

**Comment by BillWagner** [comment]
Thanks for the comments.  I updated the topic based on the feedback. Please let me know if this is ready. 

/cc @MichaelSimons @spboyer

**Comment by MichaelSimons** [comment]
:shipit:

**Pull request merged**